File: 1.two-sum.py
#
# @lc app=leetcode id=1 lang=python
#
# [1] Two Sum
#

# @lc code=start
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        mydict = {}
        for idx in range(len(nums)):     
            # print(mydict.keys())       
            if nums[idx] in mydict.keys():
                return [mydict[nums[idx]], idx]
            else:
                mydict[target - nums[idx]] = idx
                
# @lc code=end

File: 1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> mymap;
        vector<int> ans;
        for(int i=0; i < nums.size(); i++){
            if(mymap.find(nums[i]) != mymap.end()){ // found
                ans.push_back(mymap[nums[i]]);
                ans.push_back(i);
                return ans;
            }
            else{
                mymap[target - nums[i]] = i;
            }
        }
        return ans;      
    }
};
// @lc code=end

